changelogs/current.yaml

```diff
@@ -22,6 +22,11 @@ behavior_changes:
     Fixed tag extraction so that :ref:`stat_prefix <envoy_v3_api_msg_extensions.filters.udp.dns_filter.v3.DnsFilterConfig>`
     is properly extracted. This changes the Prometheus name from
     dns_filter_myprefix_local_a_record_answers{} to dns_filter_local_a_record_answers{envoy.dns_filter_prefix="myprefix"}.
+- area: stats connection_limit
+  change: |
+    Fixed tag extraction so that :ref:`stat_prefix <envoy_v3_api_msg_extensions.filters.network.connection_limit.v3.ConnectionLimit>`
+    is properly extracted. This changes the Prometheus name from envoy_connection_limit_myprefix_limited_connections{} to
+    envoy_connection_limit_limited_connections{envoy_connection_limit_prefix="myprefix"}.
```

Comment:
Did you mean:
```
envoy_connection_limit.myprefix_limited_connections{} to
envoy_connection_limit.limited_connections{envoy_connection_limit_prefix="myprefix"}.
```
Based on your test it looks like it to me.

Reply:
According to [Prometheus data model](https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels), metric names are not allowed to contain dots.  So, `PrometheusStatsFormatter::metricName` replaces dots with underscores.
https://github.com/envoyproxy/envoy/blob/21323d483ed163155fb8ceba0935c33c6dae7a2f/source/server/admin/prometheus_stats.cc#L311
https://github.com/envoyproxy/envoy/blob/21323d483ed163155fb8ceba0935c33c6dae7a2f/source/server/admin/prometheus_stats.cc#L24-L29

I believe the following is accurate.
```
``envoy_connection_limit_myprefix_limited_connections{}`` to
``envoy_connection_limit_limited_connections{envoy_connection_limit_prefix="myprefix"}``.
```